Newsletters: add 162 (2021-08-18) (#628)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2021-08-18-newsletter.md

release candidates.*

- [Bitcoin Core 22.0rc2][bitcoin core 22.0] is a release candidate
  for the next major version of this full node implementation and its
  associated wallet and other software. Major changes in this new
  version include support for [I2P][topic anonymity networks] connections,
  removal of support for [version 2 Tor][topic anonymity networks] connections,
  and enhanced support for hardware wallets.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core #22642][] updates Bitcoin Core's release process for
  upcoming version 22.0 to concatenate the GPG signatures of everyone
  who [reproducibly built][topic reproducible builds] the binaries into a single file which
  can be batch verified ([example][gpg batch]).  Signatures from

>   determinsitic builders have been available for years, but this should

Reviewer: ```suggestion
  deterministic builders have been available for years, but this should
```

  make them more accessible and also reduce the existing dependency
  on the project's lead maintainer signing the release binaries.

- [Bitcoin Core #21800][] mempool/validation: mempool ancestor/descendant limits for packages FIXME:jnewbery

- [Bitcoin Core #21500][] updates the `listdescriptors` RPC with a
  `private` parameter that, when set, will return the private form of
  each descriptor.  The private form contains any known private keys or
  extended private keys (xprvs), allowing this updated command to be
  used to backup the wallet.

- [Rust-Lightning #1009][] adds a `max_dust_htlc_exposure_msat` channel
  configuration option which limits the total balance of pending "dusty HTLCs"
  whose amounts are below the dust limit.

  This change is in preparation for a [proposed][BOLTs #873]
  `option_dusty_htlcs_uncounted` feature bit, which advertises that the node
  does not wish to count "dusty HTLCs" against `max_accepted_htlcs`. Node
  operators would likely want to adopt this feature bit since
  `max_accepted_htlcs` is mainly used to limit the potential size of the onchain
  transaction if a force-close were to happen and "dusty HTLCs" are unclaimable

>   onchain and would never affect the final transction size.

Reviewer 2: ```suggestion
  onchain and would never affect the final transaction size.
```

Reviewer 3: Fixed!

  version include support for [I2P][topic anonymity networks] connections,
  removal of support for [version 2 Tor][topic anonymity networks] connections,
  and enhanced support for hardware wallets.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core #22642][] updates Bitcoin Core's release process for
  upcoming version 22.0 to concatenate the GPG signatures of everyone
  who [reproducibly built][topic reproducible builds] the binaries into a single file which
  can be batch verified ([example][gpg batch]).  Signatures from
  determinsitic builders have been available for years, but this should
  make them more accessible and also reduce the existing dependency
  on the project's lead maintainer signing the release binaries.

> - [Bitcoin Core #21800][] implements ancestor and descendant limits for
>   mempool package acceptance. Bitcoin Core limits the number of
>   related transactions in its mempool as a protection against DoS
>   attacks and so that block construction is tractable for miners. By default,
>   those [limits][bitcoin core mempool limits] ensure that no transaction in
>   the mempool can have  more than 25 ancestors or descendants, and that no
>   transaction can have ancestors or descendants that have a combined weight of
>   101kVb.

Reviewer 4: The limits are inclusive of the tx itself. Also, it's kind of hard to word this in a clear way, but I hope to articulate that there are a total of 4 limits:

```suggestion
- [Bitcoin Core #21800][] implements ancestor and descendant limits for
  mempool package acceptance. Bitcoin Core limits the number of
  related transactions in its mempool as a protection against DoS
  attacks and so that block construction is tractable for miners. By default,
  those [limits][bitcoin core mempool limits] ensure that no transaction
  in the mempool, combined with its mempool ancestors, can exceed 25
  transactions or 101KvB in weight. The same rules apply to the transaction
  combined with its mempool descendants.
```

    Those ancestor and descendant limits are enforced when a transaction is
    considered for addition to the mempool. If adding the transaction would
    cause one of those limits to be exceeded, then the transaction is rejected.

>     [#21800][bitcoin core #21800] adds ancestor and descendant limit checks for
>     mempool package acceptance (i.e. when multiple transactions are considered
>     for addition to the mempool at the same time). Mempool package acceptance

Reviewer 4: The main thing about #21800 is it's safe for any arbitrary package. I'll optimize it in future PRs when stuff is more specific (also I think harding mentioned something about looking for feedback on package limits?) - I'm pretty sure it's going to be 2 generation, multi parent + 1 child, but probably good idea to say it's not finalized and maybe interested people will know where to make suggestions?

```suggestion
    While package semantics have not been finalized, [#21800][bitcoin core #21800]
    implements ancestor and descendant limit checks for validating
    arbitrary packages (i.e. when multiple transactions are considered
    for addition to the mempool at the same time). Mempool package acceptance
```
